log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+    };
+    scales.push(colorScale);
+    marks.push(getPieMark(config, this.metadata));
+
+    if (config.percentage) {
+        marks.push(getPieText(config, this.metadata));
+    }
+
+
+    var legendTitle = "Legend";
+
+    if (config.title != "table") {
+        legendTitle = config.title;
+    }
+
+    var legends = [
+        {
+            "fill": "color",
+            "title": "Legend",
+            "offset": 20,
+            "properties": {
+                "symbols": {
+                    "fillOpacity": {"value": 0.5},
+                    "stroke": {"value": "transparent"}
+                }
+            }
+        }
+    ];
+
+    this.spec.legends = legends;
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+};
+
+arc.prototype.draw = function(div, callbacks) {
+
+    var viewUpdateFunction = (function(chart) {
+        if(this.config.tooltip.enabled){
+            this.config.tooltip.type = "arc";
+            createTooltip(div);
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+            bindTooltip(div,this.view,this.config,this.metadata);
+        } else {
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+        }
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
+            var allowedDataSet = [];
+            var startingPoint = dataset.length - maxValue;
+            for(var i = startingPoint; i < dataset.length;i++){
+                allowedDataSet.push(dataset[i]);
+            }
+            this.spec.data[0].values = allowedDataSet;
+        }
+    }
+
+    vg.parse.spec(this.spec, viewUpdateFunction);
+};
+
+arc.prototype.insert = function(data) {
+
+    var color = this.metadata.names[this.config.color];
+    var x = this.metadata.names[this.config.x];
+    var view =this.view;
+
+
+
+    var updated = false;
+
+    for (i = 0; i < data.length; i++) {
+        this.view.data(this.config.title)
+            .update(
+                function(d) {
+                    return d[color] == data[i][color];
+                },
+                x,
+                function(d) {
+                    updated = true;
+                    return data[i][x];
+                });
+    }
+
+    if (updated == false) {
+        view.data(this.config.title).insert(data);
+    }
+
+    this.view.update({duration: 500});
+};
+
+arc.prototype.getSpec = function() {
+    return this.spec;
+};
+
+
+function getPieMark(config, metadata){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD